### PR TITLE
feat: k8s CRDs & integration tests

### LINKED ISSUE #631: k8s backend: error creating CRD
```
../../akashctl --home cache/client  --keyring-backend=test provider run \
	--from provider \
	--cluster-k8s
ERROR: kube: error creating akashv1 CRD: CustomResourceDefinition.apiextensions.k8s.io "manifests.akash.network" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required
```

### LINKED ISSUE #632: _run/single: deploy apps to k8s
We should be able to deploy apps to k8s with this example.

* [ ] create provider
* [ ] run provider
* [ ] generate lease
* [ ] push manifest
* [ ] ensure app deployed

### LINKED ISSUE #633: k8s: invalid namespaces names for leases
new namespaces are too long; 63-char max.

```
E[2020-06-02|15:29:26.541] applying namespace                           module=provider-cluster-kube err="Namespace \"f996e45559d39e30e32266bfbf77b4a8d636541087f7111f285cdc3f9b5a1c5c\" is invalid: metadata.name: Invalid value: \"f996e45559d39e30e32266bfbf77b4a8d636541087f7111f285cdc3f9b5a1c5c\": must be no more than 63 characters" lease="{Owner:1F9421CC0DD2FD6B6B3C7261F76312F044784722 DSeq:620 GSeq:1 OSeq:1 Provider:C09CA2FA4E0D8ED26E40109CDE03177399755A3F}"
```

### PR DESCRIPTION
**Pull request by boz**
* refactor/fix CRDs
* test crds on CI
* testutil: manifest generators
* types: use uint16 for ports
* TODO: `provider/cluster/kube` k8s-integration tests

fixes #631
refs  #632
fixes #633

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -166,13 +183,14 @@ clean:
...
 	test-lint lintdeps-install \
+	test-k8s-integration \
 	test-vet \
 	vendor \
 	mocks \
 	gofmt \
 	docs \
 	clean \
-	kubetypes \
+	kubetypes kubetypes-deps-instal \
```
Comment (troian) [inline Makefile:193]: `kubetypes-deps-install`

File: testutil/manifest_rand.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	t.Helper()
+	return []manifest.Group{
+		mg.Group(t),
+	}
+}
+
+func (mg manifestGeneratorRand) Group(t testing.TB) manifest.Group {
+	t.Helper()
+	return manifest.Group{
+		Name: "left-coast",
```
Comment (Ropes) [inline testutil/manifest_rand.go:23]: nice!

File: script/kind-metrics-server.yaml [hunk 3]
```diff
@@ -0,0 +1,153 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:aggregated-metrics-reader
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
```
[the remark this answers is not in the document]
Comment (boz) [reply script/kind-metrics-server.yaml:9]: just pulling down [what is suggested](https://github.com/ovrclk/akash/pull/639/files#diff-31ba2edfa4795b4c80e564caf1c7cce9R16) for it to work in kind.

File: _run/kube/Makefile [hunk 4]
```diff
@@ -39,11 +39,19 @@ init-keys: dirs
...
 	$(akashd) start
 
 run-provider:
-	$(akashctl) $(KEY_OPTS) --from provider  provider run
+	$(akashctl) $(KEY_OPTS) provider run \
+		--from provider \
+		--cluster-k8s
+
+provider-status:
+	curl -s http://localhost:8080/status | jq
```
[the remark this answers is not in the document]
Comment (boz) [reply _run/kube/Makefile:54]: :+1: I'll add it when I flush this one out in the next iteration.

File: script/kind-metrics-server.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (boz) [reply script/kind-metrics-server.yaml:9]: also see #609

**Comment by boz** [comment]
@Ropes I didn't want to write the wait loop in a make target.

**Pull request merged**